code/obj/item/assembly/misc_assemblies.dm

```diff
@@ -920,7 +943,10 @@ obj/item/assembly/radio_horn/receive_signal()
 	attackby(obj/item/W, mob/user)
 		if (!recipe) //no recipie? assign one
 			if (istype(W, /obj/item/raw_material/shard))
-				recipe = new/datum/pipeshotrecipe/glass
+				if(W.material.getProperty("hard") >= 7)
```

Comment:
This is a really inaccurate check, since you're now just looking at any crystal shard that has hardness greater than 7, e.x. uqill, custom crystal mats, etc.

Why not just use a `mat_id` check instead? See: https://github.com/goonstation/goonstation/blob/ab1e081d435f29a359ce391e96e639427ccafff0/code/modules/materials/Mat_Materials.dm#L658

Reply:
The reason I didn't do that is to prevent folks from alloying plasmaglass to get more shards, but on second thought, there's no reason it shouldn't use plasmaglass